### PR TITLE
Hexa mesh generator

### PR DESCRIPTION
**Pull request by bmxam**
Helpfull for debug / prototyping

### CODE REVIEW

File: src/mesh/mesh_generator.jl [hunk 1]
```diff
@@ -72,15 +72,18 @@ Generate either a line mesh, a rectangle mesh, a cubic mesh... depending on the
 
 # Example
 ```julia-repl
-mesh_of_a_line   = ncube_mesh([10])
-mesh_of_a_square = ncube_mesh([4, 5])
+mesh_of_a_line = ncube_mesh([10])
+mesh_of_a_square = ncube_mesh([4, 5]
```
Comment (ghislainb) [inline src/mesh/mesh_generator.jl:76]: missing `)`

File: src/mesh/mesh_generator.jl [hunk 2]
```diff
@@ -364,6 +367,104 @@ function _rectangle_quad_mesh(nx, ny, xmin, xmax, ymin, ymax, ::Val{2}, bnd_name
...
+                    (iz - 1 + 1) * nx * ny + (iy - 1 + 0) * nx + ix + 0
+                cell2node[8 * ielt - 2] =
+                    (iz - 1 + 1) * nx * ny + (iy - 1 + 0) * nx + ix + 1
+                cell2node[8 * ielt - 1] =
+                    (iz - 1 + 1) * nx * ny + (iy - 1 + 1) * nx + ix + 1
+                cell2node[8 * ielt - 0] =
+                    (iz - 1 + 1) * nx * ny + (iy - 1 + 1) * nx + ix + 0
+            end
+        end
+    end
```
Comment (ghislainb) [inline src/mesh/mesh_generator.jl:451]: could you try the following ? 
```julia
cell2node = zeros(Int, 8, nelts)
ijk_nodes = CartesianIndices((1:nx, 1:ny, 1:nz))
id_nodes = LinearIndices(ijk_nodes)
ijk_cells = CartesianIndices((1:(nx - 1), 1:(ny - 1), 1:(nz - 1)))
id_cells = LinearIndices(ijk_cells)
node_order = SA[1, 2, 4, 3, 5, 6, 8, 7]
stencilcube = CartesianIndices((0:1, 0:1, 0:1))[node_order]
for ic in id_cells
    cell2node[:, ic] .= id_nodes[ijk_cells[ic] .+ stencilcube]
end
cell2node = vec(cell2node)
```

Comment (bmxam) [reply src/mesh/mesh_generator.jl:451]: I'd rather not to. I thought of it too, but I'd like use to do this in a separate PRs, where we could generate line, rectangle and hexa mesh with the same route (using Linear and Cartesian indexing). But for now I just need this simple hexa mesh for dev purpose so let's improve this later.

Comment (ghislainb) [reply src/mesh/mesh_generator.jl:451]: Ok, I can do that latter

**Review by ghislainb — APPROVED** [review]

**Pull request merged**